core/src/main/java/hudson/security/Permission.java

```diff
@@ -222,7 +225,7 @@ public boolean isContainedBy(@Nonnull PermissionScope s) {
      * @see #fromId(String)
      */
     public @Nonnull String getId() {
-        return owner.getName()+'.'+name;
+        return id;
```

Comment:
Are Permission objects ever serialized in configuration?  Generally they're statically created AFAIK but if a plugin lets you select the permission type to apply to some property (none I know of off the top of my head) then we could have cases where this is null. 

If so, we should probably check if field is non-null and fall back to old logic when null.